src/celeritas/decay/data/MuDecayData.hh

```diff
@@ -0,0 +1,64 @@
+//----------------------------------*-C++-*----------------------------------//
+// Copyright 2024 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/decay/data/MuDecayData.hh
+//---------------------------------------------------------------------------//
+#pragma once
+
+#include "corecel/Macros.hh"
+#include "corecel/Types.hh"
+#include "celeritas/Quantities.hh"
+#include "celeritas/Types.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Data storage for muon decay.
+ */
+struct MuDecayIds
+{
+    ParticleId mu_minus;
+    ParticleId mu_plus;
+    ParticleId electron;
+    ParticleId positron;
+
+    //! Check whether the data is assigned
+    explicit CELER_FUNCTION operator bool() const
+    {
+        return mu_minus && mu_plus && electron && positron;
+    }
+};
+
+//---------------------------------------------------------------------------//
+/*!
+ * Device data for creating a muon decay interactor.
+ */
+struct MuDecayData
+{
+    //!@{
+    //! \name Type aliases
+    using Mass = units::MevMass;
+    //!@}
+
+    //! Particle identifiers
+    MuDecayIds ids;
```

Comment:
We don't really need separate IDs for these any more: that used to be a limitation of how we were instantiating the interactors. It might improve readability a bit through: `data.ids.positron` instead of `data.positron_id`?